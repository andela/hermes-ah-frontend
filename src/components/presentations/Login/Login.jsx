import { Form, Button } from 'semantic-ui-react';
import { Link, Redirect } from 'react-router-dom';
import React from 'react';
import FormComponent from '../../shared/Form/Form';
import Loader from '../../shared/Loader/Loader';

class Login extends FormComponent {
  constructor(props) {
    super(props);
    this.state = {
      data: { email: '', password: '' },
    };
  }

  handelClick = async e => {
    e.preventDefault();
    const { data } = this.state;
    const { login: loginUser } = this.props;
    await loginUser(data);
  };

  render() {
    const emailIdType = 'email';
    const emailPlaceHolder = 'Email';
    const passwordIdType = 'password';
    const passwordPlaceHolder = 'Password';
    const { user, isLoadingReducer } = this.props;
    const { loader } = isLoadingReducer;
    if (user) return <Redirect to="/" />;
    return (
      <div>
        {loader && <Loader />}
        <React.Fragment>
          <div className="form-wrap">
            <div className="form-cont">
              <div className="wrap-cont-login-signup">
                <div className="button-container">
                  <Button className="login current-login">LOGIN</Button>
                </div>
                <div className="button-container">
                  <Link to="/signup">
                    <Button className="signup">SIGNUP</Button>
                  </Link>
                </div>
              </div>
              <Form unstackable onSubmit={this.handelClick}>
                {this.renderInput(emailIdType, emailPlaceHolder, emailIdType)}
                {this.renderInput(
                  passwordIdType,
                  passwordPlaceHolder,
                  passwordIdType
                )}
                <Link to="/forgot-password" className="forgot-password">
                  Forgot password?
                </Link>
                <Button className="submit" type="submit">
                  LOGIN
                </Button>
                {this.renderRuler()}
              </Form>
              {this.renderSocialLogin()}
            </div>
          </div>
        </React.Fragment>
      </div>
    );
  }
}

export default Login;
