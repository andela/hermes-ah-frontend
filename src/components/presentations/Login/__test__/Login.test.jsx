import React from 'react';
import Login from '../Login';

describe('Login component', () => {
  it('should match snapshot', () => {
    const wrap = <Login />;
    expect(wrap).toMatchSnapshot();
  });
});
