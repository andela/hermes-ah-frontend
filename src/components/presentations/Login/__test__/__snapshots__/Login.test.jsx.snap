// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `<Login />`;
